youtube_dl/extractor/msn.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..utils import (
+    unescapeHTML,
+    int_or_none,
+)
+
+class MSNIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?msn\.com/.+/[a-z]{2}-(?P<id>[a-zA-Z]+)'
+    _TESTS = [{
+        'url': 'http://www.msn.com/en-ae/foodanddrink/joinourtable/criminal-minds-shemar-moore-shares-a-touching-goodbye-message/vp-BBqQYNE',
+        'md5': '8442f66c116cbab1ff7098f986983458',
+        'info_dict': {
+            'id': 'BBqQYNE',
+            'title': 'Criminal Minds - Shemar Moore Shares A Touching Goodbye Message',
+            'description': 'Following news of his departure from the show, the star of Criminal Minds shares a heartwarming message to his fans and "baby girls."',
+            'duration': 104,
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'http://www.msn.com/en-ae/news/offbeat/meet-the-nine-year-old-self-made-millionaire/ar-BBt6ZKf',
+        'md5': 'bfbb6edbc806d34218c7f326d3dddd74',
+        'info_dict': {
+            'id': 'BBt6ZKf',
+            'title': 'All That Bling: Self-Made Millionaire Child Builds Fashion &amp; Jewellery Empire',
+            'description': 'md5:8e683bd5c729d5fb16d96539a582aa5e',
+            'duration': '350',
+            'ext': 'mp4',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        self.report_extraction(video_id)
+        video_data = self._parse_json(unescapeHTML(self._search_regex(r'data-metadata\s*=\s*["\'](.+)["\']',
+            webpage, 'video data')), video_id)
+
+        subtitles = {}
+        for f in video_data.get('files', []):
+            if f.get('formatCode', '') == '3100':
+                lang = f.get('culture', '')
+                if not lang:
+                    continue
+                subtitles[lang] = []
+                subtitles[lang].append({
+                    'ext': 'ttml',
+                    'url': f.get('url', '').replace('&amp;', '&'),
+                })
+
+        formats = []
+        for video_file in video_data.get('videoFiles', []):
+            if not '.ism' in video_file.get('url', '.ism'):
```

Comment:
In the give example URL there are two ```.ism``` URLs. The first one
```
http://wus.streaming.video.msn.com/806b30dc-9570-485a-b278-4b8111d4982b/a9a2473c-ff61-45d6-90b8-cac019422d6c.ism
```
Is not a valid URL - need to append ```/manifest```. The resulting URL
```
http://wus.streaming.video.msn.com/806b30dc-9570-485a-b278-4b8111d4982b/a9a2473c-ff61-45d6-90b8-cac019422d6c.ism/manifest
```
Is a Microsoft Smooth Streaming (#8118) manifest. As Microsoft Smooth Streaming is not implemented yet and it's not required to support MSN, you can ignore it and add some comments. The other one:
```
http://wus.streaming.video.msn.com/806b30dc-9570-485a-b278-4b8111d4982b/a9a2473c-ff61-45d6-90b8-cac019422d6c.ism/manifest(format=m3u8-aapl)
```
Is a valid M3U8 manifest and should be extracted.